api/middleware/globalErrorCatcher.js

```diff
@@ -0,0 +1,3 @@
+export const globalErrorCatcher = (handler) => async (req, res, next) => {
+    return await handler(req, res, next)?.catch(next);       
```

Comment:
r u sure this will work? should it not wrap ur code using a try catch? something like this?

`try {
  return await handler(req, res, next)
} catch(error){
  next(new Error(error))
}`

Reply:
Yeah mine works